presto-native-execution/presto_cpp/main/PrestoTask.h

```diff
@@ -95,8 +95,14 @@ struct PrestoTask {
   /// has not been started, until the actual 'create task' message comes.
   bool taskStarted{false};
 
+  /// Time point (in ms) when the last message (any) came for this task.
   uint64_t lastHeartbeatMs{0};
```

Comment:
```
s/lastHeartbeatMs/lastWorkerHeartbeatMs/
```
Do we need to track the last heartbeat or message received from worker? Or shall we rename it to lastWorkerMsgTimeMs?

Reply:
I don't think we should do this renaming in this change.
It is unnecessary and is actually unclear why you are suggesting it.

> Do we need to track the last heartbeat or message received from worker? Or shall we rename it to lastWorkerMsgTimeMs?

If we need this - it would be a separate PR.

Reply:
I am wondering if we need to update lastHeartbeatMs when we receive message from worker? Or we only need to record the heartbeat timestamp from coordinator? Thanks!

Reply:
> I am wondering if we need to update lastHeartbeatMs when we receive message from worker? Or we only need to record the heartbeat timestamp from coordinator? Thanks!

I don't know that - just keeping the existing heartbeat as a status quo in order not to break the current behavior.

What I can do is to follow up by looking at the java code and see if it has something similar and how it is updated.
Not sure what I will have time for it, because it is low pri.